### ISSUE TITLE
Implement macros for PEP 518 & PEP 517 build systems

### ISSUE THREAD

**Issue by mcepl** [openSUSE/python-rpm-macros#30]
See https://bugzilla.redhat.com/1685582 and there particularly links to https://discuss.python.org/t/pep-517-build-frontends-to-inform-users-about-wheel-basename/1026 and https://github.com/pypa/pip/pull/6377

The first draft of proposed macros is
```
%pyproject_wheel: <flags...> %{__python3} -m pip wheel --no-deps --use-pep517 --no-build-isolation --progress-bar off --verbose .
+ conditional --global-option and --build-option based on %{?py_setup_args} and/or other macros.

%pyproject_install: %{__python3} -m pip install --root %{buildroot}  --strip-file-prefix %{buildroot} --no-deps  --progress-bar off *.whl
```

**Comment by sebix** [comment]
See also https://github.com/python-poetry/poetry/issues/1338#issuecomment-657860111 poetry will drop setup.py soon and therefore all packages using poetry will not be installable with the standard way then.

**Comment by bnavigator** [comment]
The macros are in use and should be documented. #59

**Comment by mcepl** [comment]
I think the basic support of this has been included at least in 845439fe1e2d45408c122950c2cfa1bade04784b.
